bodhi/models/models.py

```diff
@@ -1649,19 +1649,35 @@ def check_requirements(self, session, settings):
 
         return True, "All checks pass."
 
+    def negative_karma(self):
+        """Checks for negative karma, Returns True if the update has negative karma"""
+        feedback = defaultdict(int)
+        for comment in self.comments:
+            if not comment.anonymous:
+                feedback[comment.user.name] +=  comment.karma
+        for karma in feedback.values():
+            if karma < 0:
+                return True
+
     def check_karma_thresholds(self, db, agent):
         """Check if we have reached either karma threshold, and call set_request if necessary"""
         if not self.locked:
             if self.status is UpdateStatus.testing:
                 if self.stable_karma not in (0, None) and self.karma >= self.stable_karma:
                     if self.autokarma:
-                        log.info("Automatically marking %s as stable" % self.title)
-                        self.set_request(db, UpdateRequest.stable, agent)
-                        self.request = UpdateRequest.stable
-                        self.date_pushed = None
-                        notifications.publish(
-                            topic='update.karma.threshold.reach',
-                            msg=dict(update=self, status='stable'))
+                        if self.critpath and self.negative_karma:
+                            log.info("Disabling Auto Push since the critic update has negative karma")
+                            self.autokarma = False
+                            self.status = UpdateStatus.testing
+                            self.request = None
```

Comment:
My familiarity with Bodhi is not strong yet, but it is not obvious to me that self gets saved to the database after these values are adjusted. In the else block, I do see the call to self.set_request(), and then some attribute changes similar to what is seen here. Do you know where these values are used and/or set into the database? I really need to sit down and just read all the Bodhi code soon!